pypeit/core/telluric.py

```diff
@@ -1718,199 +1764,377 @@ def poly_telluric(spec1dfile, telgridfile, telloutfile, outfile, z_obj=0.0, func
+            TODO: This explanation is unclear.
+        debug (:obj:`bool`, optional):
+            If True, QA plots will be shown to the screen indicating the
+            quality of the fits. Specifically, the residual distributions
+            will be shown at each iteration, and the fit will be shown at the
+            end (for each order). This is useful if you are running the code
+            for the first time, but since the algorithm is slow, particularly
+            for fitting multi-order echelle data, it will require lots of
+            clicking to close interactive matplotlib windows which halt code
+            flow.
+
     """
-    def __init__(self, wave, flux, ivar, mask, telgridfile, obj_params, init_obj_model, eval_obj_model,
-                 ech_orders=None,
-                 sn_clip=30.0, airmass_guess=1.5, resln_guess=None,
-                 resln_frac_bounds=(0.5, 1.5), pix_shift_bounds=(-5.0, 5.0), pix_stretch_bounds=(0.9,1.1),
-                 maxiter=2, sticky=True, lower=3.0, upper=3.0,
-                 seed=777, ballsize = 5e-4, tol=1e-3, diff_evol_maxiter=1000,  popsize=30, recombination=0.7, polish=True, disp=False, sensfunc=False,
-                 debug=False):
+    version = '1.0.0'
+    """Datamodel version."""
+
+    datamodel = {'telgrid': dict(otype=str,
+                                 descr='File containing grid of HITRAN atmosphere models'),
+                 'std_src': dict(otype=str, descr='Name of the standard source'),
+                 'std_name': dict(otype=str, descr='Type of standard source'),
+                 'std_cal': dict(otype=str,
+                                 descr='File name (or shorthand) with the standard flux data'),
+                 'func': dict(otype=str, descr='Polynomial function used'),
+                 'pca_file': dict(otype=str, descr='Name of the QSO PCA file'),
+                 'tol': dict(otype=float,
+                             descr='Relative tolerance for converage of the differential '
+                                   'evolution optimization.'),
+                 'popsize': dict(otype=int,
+                                 descr='A multiplier for setting the total population size for '
+                                       'the differential evolution optimization.'),
+                 'recombination': dict(otype=float,
+                                       descr='The recombination constant for the differential '
+                                             'evolution optimization. Should be in the range '
+                                             '[0, 1].'),
+                 'polish': dict(otype=bool,
+                                descr='Perform a final optimization to tweak the best solution; '
+                                      'see scipy.optimize.differential_evolution.'),
+                 'airmass': dict(otype=float, descr='Airmass of the observation'),
+                 'exptime': dict(otype=float, descr='Exposure time'),
```

Comment:
Add  `(s)`

Reply:
Done ... and assuming it's always in seconds.